Helper/Webhook/RecurringContractWebhookHandler.php

```diff
@@ -101,4 +77,28 @@ private function handleFailedNotification(MagentoOrder $order, Notification $not
             );
         }
     }
+
+    private function handleSuccessNotification(MagentoOrder $order, Notification $notification): void
+    {
+        $paymentMethodInstance = $order->getPayment()->getMethodInstance();
+        $paymentMethodCode = $paymentMethodInstance->getCode();
+        $storeId = $paymentMethodInstance->getStore();
+        $isRecurringEnabled = $this->vaultHelper->getPaymentMethodRecurringActive($paymentMethodCode, $storeId);
+
+        if ($isRecurringEnabled) {
+            try {
+                $paymentToken = $this->vaultHelper->createVaultToken($order->getPayment(), $notification->getPspreference());
+                $extensionAttributes = $this->vaultHelper->getExtensionAttributes($order->getPayment());
+                $extensionAttributes->setVaultPaymentToken($paymentToken);
+            } catch (Exception $exception) {
+                $this->adyenLogger->error(
+                    sprintf(
+                        'Failure trying to save payment token in vault for order %s, with exception message %s',
+                        $order->getPayment()->getOrder()->getIncrementId(),
+                        $exception->getMessage()
+                    )
+                );
+            }
+        }
```

Comment:
Shall we add an `else` block for `isRecurringEnabled === false` case and log a message?